src/main/webapp/user_profile_script.js

```diff
@@ -52,10 +104,105 @@ function closeModal() {
     modal.style.display = "none";
 }
 
+function closeEditModal() {
+    var modal = document.getElementById("editTaskModal");
+    modal.style.display = "none";
+}
+
 // If the user clicks outside of the modal, closes the modal directly
 window.onclick = function(event) {
     var modal = document.getElementById("createTaskModal");
     if (event.target == modal) {
         modal.style.display = "none";
     }
+    var editModal = document.getElementById("editTaskModal");
+    if (event.target == editModal) {
+        editModal.style.display = "none";
+    }
+}
+
+async function displayNeedHelpTasks() {
+    const queryURL = "mytasks?keyword=Owner";
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const taskResponse = await response.json();
+    const needHelpTable = document.getElementById("need-help");
+    needHelpTable.innerHTML = "";
+    var headerRow = document.createElement("tr");
+    var headers = ["Task Overview", "Helper", "Status", "Edit", "Delete"];
+    for (var index = 0; index < headers.length; index++) {
+        var th = document.createElement("th");
+        th.appendChild(document.createTextNode(headers[index]));
+        headerRow.appendChild(th);
+    }
+    needHelpTable.appendChild(headerRow);
+    for (var index = 0; index < taskResponse.length; index++) {
+        var tr = document.createElement("tr");
+        var task = taskResponse[index];
+        var data = [task.detail, task.helper, task.status];
+        for (var i = 0; i < data.length; i++) {
+            var td = document.createElement("td");
+            td.appendChild(document.createTextNode(data[i]));
+            tr.appendChild(td);
+        }
+        const keyStringCopy = task.keyString.slice();
```

Comment:
What's the purpose of this copy?

Reply:
At first I didn't make a copy for the keyString. But then I found that for all the onclick functions I set for the edit buttons, when the button is clicked, the function editTask(keyString) should be called and the keyString should be different for each onclick function. However, if I don't make a copy of the keyString, all the editTask function will take in the same keyString input. Thus, it seems that without making a copy, all the input keyString would just be aliases referring to the same keyString. As a result, I make a copy for each separate keyString.

Reply:
Ah, yes it probably has to do with the anonymous function on line 155 that was referring to the loop variable `index`. Keeping this copy SGTM.